versioned/mongodb/src/main/java/com/dremio/nessie/versioned/impl/condition/BsonConditionExpression.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dremio.nessie.versioned.impl.condition;
+
+import org.bson.conversions.Bson;
+
+/**
+ * This provides a separation of queries on @{ConditionExpression} from the object itself.
+ * This uses the Visitor design pattern to retrieve object attributes.
+ */
+public class BsonConditionExpression implements ConditionExpressionVisitor<Bson> {
+
+  /**
+   * Creates a BSON representation of the ConditionExpression object.
+   * @param conditionExpression the object to convert.
+   * @return BSON representation of conditionExpression
+   */
+  public Bson to(ConditionExpression conditionExpression) {
+    return conditionExpression.accept(this);
+  }
+
+   /**
+   * This is a callback method that ConditionExpression will call when this visitor is accepted.
+   * Its purpose is c reates a BSON representation of the ConditionExpression object.
+   * @param conditionExpression
+   * @return
+   */
+  @Override
+  public Bson visitTo(ConditionExpression conditionExpression) {
+//    final ConditionExpression conditionExpression = _conditionExpression.accept(this);
+    BsonExpressionFunction bsonExpressionFunction = new BsonExpressionFunction();
+    ExpressionFunction expressionFunction = conditionExpression.getFunctions().get(0);
```

Comment:
This should instead be something like:
ExpressionFunctionVisitor visitor = new BsonExpressionFunctionVisitor();
return conditionExpression.getFunctions().stream().map(f -> f.visit(visitor)).collect(Collectiors.toList());

Or something where you AND everything together.

Reply:
Note the `AND` above would be dependent on the operator that the ConditionExpression is.

Reply:
I have implement the stream instead of getting the first map element.
Not sure about the second comment. Do you mean `AND` is dependent the operator of the ExpressionFunction, which is what ConditionExpression comprises? I think they should be independent but willing a to accept I may have missed something in my understanding.

Reply:
No - it seems my understanding was incorrect, multiple elements in a ConditionExpression are implicitly ANDed so as long as that's happening.